autopush/db.py

```diff
@@ -424,19 +431,20 @@ def delete_messages_for_channel(self, uaid, channel_id):
         chidmessageids = [x["chidmessageid"] for x in results]
         if chidmessageids:
             self.delete_messages(uaid, chidmessageids)
+        return len(chidmessageids) > 0
 
     @track_provisioned
-    def delete_all_for_user(self, uaid):
+    def delete_user(self, uaid):
         """Deletes all messages and channel info for a given uaid"""
         results = self.table.query_2(
             uaid__eq=uaid,
-            chidmessageid__gt=" ",
+            chidmessageid__gte=" ",
             consistent=True,
             attributes=("chidmessageid",),
         )
         chidmessageids = [x["chidmessageid"] for x in results]
         if chidmessageids:
-            self.delete_messages(uaid, chidmessageids)
+            self.delete_messages(uaid, chidmessageids, True)
```

Comment:
and now I remember. The problem is that the query in deleteUser returns None as the chidmessageid for " ". This means that the " " isn't included in the list and is left behind, meaning that the UAID is never completely removed from the database.